Tests/AppTests/Helpers/QueueIsolated.swift

```diff
@@ -22,31 +22,39 @@ import Foundation
 public final class QueueIsolated<Value> {
     private var _queue = DispatchQueue(label: "queue-isolated")
 
-    public var value: Value
+    private var _value: Value
 
     public init(_ value: Value) {
-        self.value = value
+        self._value = value
+    }
+
+    public var value: Value {
+        get {
+            _queue.sync {
+                return self._value
+            }
+        }
     }
```

Comment:
I _think_ this was a real problem. As opposed to the `IsolatedActor`, which I modelled this after, plain access to the property was _not_ guarded by the queue, while on `IsolatedActor` the plain property _is_ guarded by the actor.

I believe that was the mistake and wrapping the property in the queue should fix this.

I.e. this access which is the failing assert:

```swift
        XCTAssertEqual(commands.value.count, 34)
```

raced with the `setValue` in the closure. If I'd written this as

```swift
    commands.withValue {
        XCTAssertEqual($0.count, 34)
    }
```

it wouldn't have raced, either.

We need the `value` attribute accessor in other places, so I've simply fixed that properly.

Fingers crossed 🤞

I'm going to run this a fair few times in CI to see if it reappears.